check_limits.py

```diff
@@ -1,18 +1,62 @@
+limits = {
+            'temperature': {'min': 0, 'max': 45},
+            'state_of_charge': {'min': 20, 'max': 80},
+            'charge_rate': {'min': 0,'max': 0.8}
+         } 
 
-def battery_is_ok(temperature, soc, charge_rate):
-  if temperature < 0 or temperature > 45:
-    print('Temperature is out of range!')
-    return False
-  elif soc < 20 or soc > 80:
-    print('State of Charge is out of range!')
-    return False
-  elif charge_rate > 0.8:
-    print('Charge rate is out of range!')
-    return False
+error_messages               = {'low_breach'  : { 'DE' : 'Untergrenze überschritten für ', 
+                                                  'EN' : 'Lower Limit Breached for ' }   ,
+                                'low_warning' : { 'DE' : 'Warnung vor Untergrenze für '  ,   
+                                                  'EN' : 'Warning: Lower Limit approaching for '},
+                                'high_breach' : { 'DE' : 'Obergrenze überschritten für '  , 
+                                                  'EN' : 'Higher Limit Breached for ' }   ,
+                                'high_warning': { 'DE' : 'Warnung vor höherer Grenze für ', 
+                                                  'EN' : 'Warning: Higher Limit approaching for ' } }
 
-  return True
+BMS_Status = True
+def tolerance(value_tmp):
+    return ((value_tmp * 5)/100)
+ 
+def sendBMSOutput():
+    return BMS_Status;
+    
+def fahrenheitToCelsius(temp_fahernheit):
+    return ((temp_fahernheit - 32)* 5/9)
 
+def verifyParameterTolerance(parameter,maximum,minimum,Parameter_Name,lang): 
+    if(parameter < minimum+tolerance(maximum)):
+        print(error_messages['low_warning'][lang]+Parameter_Name)
+    elif(parameter > maximum-tolerance(maximum)):
+        print(error_messages['high_warning'][lang]+Parameter_Name)
+    else:
+        print(Parameter_Name+" is Normal \n")
+  
 
-if __name__ == '__main__':
-  assert(battery_is_ok(25, 70, 0.7) is True)
-  assert(battery_is_ok(50, 85, 0) is False)
+def verifyParameter(parameter,param_name,lang):
+    
+  minimum = limits[param_name]['min']
+  maximum = limits[param_name]['max']
+  if(parameter < minimum):
+    print(error_messages['low_breach'][lang]+param_name)
```

Comment:
The verification and printing happen together. See if you can separate them.
Try your hand at accumulation. Python has very good facility to do that.